src/App/ChaosRecipeEnhancer.UI/Services/FilterManipulation/FilterStorage/FileFilterStorage.cs

```diff
@@ -93,7 +94,12 @@ public async Task WriteLootFilterAsync(string filter)
             catch (IOException e) when (i < maxRetries - 1)
             {
                 _log.Error("IOException encountered: " + e.Message);
-                await Task.Delay(delayOnRetry);
+                
+                Dispatcher.CurrentDispatcher.InvokeAsync(async () =>
```

Comment:
I should probably remove here, this will actually just stop the delay working I believe?